### PR TITLE
go generics update

### PR DESCRIPTION
**Pull request by aditya520**
Implemented generics to cachestore!

### CODE REVIEW

File: memlru/memlru.go [hunk 1]
```diff
@@ -12,25 +12,25 @@ import (
 	lru "github.com/hashicorp/golang-lru"
 )
 
-var _ cachestore.Store = &MemLRU{}
+// var _ cachestore.Store = &MemLRU{}
```
Comment (pkieltyka) [inline memlru/memlru.go:15]: can we keep this?

Comment (pkieltyka) [reply memlru/memlru.go:15]: maybe try `var _ cachestore.Store = &MemLRU[any]{}` ?

File: nostore/nostore.go [hunk 2]
```diff
@@ -7,46 +7,47 @@ import (
 	"github.com/goware/cachestore"
 )
 
-var _ cachestore.Store = &NoStore{}
+// var _ cachestore.Store = &NoStore{}
```
Comment (pkieltyka) [inline nostore/nostore.go:10]: lets keep this

File: go.mod [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
-go 1.16
+go 1.18
 
 require (
 	github.com/gomodule/redigo v1.8.8
 	github.com/hashicorp/golang-lru v0.5.4
 	github.com/pkg/errors v0.9.1
 	github.com/stretchr/testify v1.7.1
-)
+	github.com/davecgh/go-spew v1.1.0 // indirect
```
[the remark this answers is not in the document]
Comment (aditya520) [reply go.mod:13]: I did, and it makes it like this in 2 separate require statements! :(

File: go.mod [hunk 4]
```diff
@@ -1,10 +1,16 @@
...
 
 require (
 	github.com/gomodule/redigo v1.8.8
 	github.com/hashicorp/golang-lru v0.5.4
 	github.com/pkg/errors v0.9.1
 	github.com/stretchr/testify v1.7.1
 )
+
+require (
+	github.com/davecgh/go-spew v1.1.0 // indirect
```
Comment (pkieltyka) [inline go.mod:13]: can you find in the source where we have go-spew use..? we should leave it..

Comment (aditya520) [reply go.mod:13]: go-spew is coming from "github.com/stretchr/testify/assert"
All 3 indirect are coming from "github.com/stretchr/testify/assert"

**Pull request merged**